### ISSUE TITLE
Namespace issue when opening vector layers

### ISSUE THREAD

**Issue by snan** [mypaint/mypaint#518]
When opening an .ora-file created by another application (i.e. not mypaint) that has vector layers, it gives an error:

```
WARNING: lib.layer.data: Failed to load u'/home/sandra/.cache/mypaint/doc.2admRf/tmp/data/layer-2.svg': 'gdk-pixbuf-error-quark: Unrecognized image file format (3)'
```

The error only appears if the svg layers are made with a weird namespace.
If svg is the default namespace in the layers, it works super fine. (So that might be a workaround for anyone else who runs into this.)

In other words,
    <svg xmlns="http://www.w3.org/2000/svg"
works fine,
    <foo:svg xmlns:foo="http://www.w3.org/2000/svg"
gives the error.
(The foo-lish documents work well in inkscape and inkview.)
(I ran into this because I was serializing into xml from another data structure and I didn't set the default namespace.)

**Comment by achadwick** [comment]
The error message is coming from `lib.layer.data.SurfaceBackedLayer.load_surface_from_pixbuf_file()`, which is just suppressing an error from `lib.pixbuf.load_from_stream()`. Ultimately it's being raised because `GdkPixbuf.PixbufLoader` is failing to handle the XML data.

This smells quite like an upstream bug too. I've searched <https://bugzilla.gnome.org/> to no avail, however.

I'd accept a patch to defaultize (coining a term) SVG data before it's loaded by GdkPixbuf. IMO, any `{http://www.w3.org/2000/svg}svg` element at the root of the .svg file should count.

It might be fixable by telling GdkPixbuf.PixbufLoader more details about the incoming data too. There's a .new_with_mime_type() and a .new_with_type() that we could play with.

Do you have a sample of one of these non-default namespaced files, if they're in common use in the nom-MyPaint world somewhere?

**Comment by snan** [comment]
Here's an example of a file where this is the only difference:

- [prefixed-squares.ora](http://idiomdrottning.org/prefixed-squares.ora)
- [default-squares.ora](http://idiomdrottning.org/default-squares.ora)

The layers in data/ open in eog, dwb, inkview and inkscape.

I'm working on [a dinky little program](http://idiomdrottning.org/svgz_to_ora.scm.txt) that I used to convert [this svgz file with layers](http://idiomdrottning.org/squares.svgz) into an ora-file where the layers are separated. That's how I ran into this bug.

Arguably this is one of the least important bugs of all time, because I just changed my conversion program to defaultize it (which I would've wanted anyway, looks better).

I'm not so sure defaultization is the way to go, seems like a workaround for something that's both unusual and better fixed upstream and/or in interfacing with gdk-pixbuf.

**Comment by snan** [comment]
It's gdk_pixbuf for sure. Inspired by [this bug over at Launchpad](https://bugs.launchpad.net/ubuntu/+source/gdk-pixbuf/+bug/926019), I made a copy of default-squares.ora (i.e. the defaultized one) that had a comment, seven or eight lines or so, before the root element on one of the layers. That layer failed to load, just as the others.

My guess is that it is not good at realizing that the file is an svg file.
So our options are `.new_with_mime_type()`, `.new_with_type()` or waiting for upstream.

**Comment by snan** [comment]
Went to report this at bugzilla.gnome.org but [they had the comment bug already](https://bugzilla.gnome.org/show_bug.cgi?id=669854), and they closed it as upstream. The issue is in shared-mime-info.

Looking around at upstream's upstream, I found that [they've closed it](https://bugs.freedesktop.org/show_bug.cgi?id=27318) and recommend using the file extension to detect the files.

...

:/

**Comment by jplloyd** [comment]
4½ short years later, this is now fixed in: 03dca81d41070f046b56c04d41dcb9174e49ef80 and eaafc10b69c536aa5d43920ac03522b22ec28056
